### PR TITLE
add repository configuration for server packages

### PR DESCRIPTION
**Pull request by mrunge**

### CODE REVIEW

**Comment by Spredzy** [comment]
@mrunge I was wondering why creating an extra role for that when this content could be integrated directly within the role they provide repository for. It is actually best practice to do that so roles can actually live on their own without the need to depend on other repo just for that.

It would be strange to have an ElasticSearch role that can't install ElasticSearch itself.

Taking Puppet ElasticSearch module as an example: https://github.com/elastic/puppet-elasticsearch/blob/master/manifests/repo.pp

**Comment by paramite** [comment]
+1 on having this in each existing role + I would add also global manage_repo which would disable repo creating in case it's necessary.

**Comment by mrunge** [comment]
Global switch or a per role switch for repositories?

**Comment by paramite** [comment]
That's a good question. I think that, given the fact we don't differentiate on manage_packages etc., the global switch is enough.

**Review by richm — APPROVED** [review]

**Comment by mrunge** [comment]
I moved grafana and elasticsearch repo config to their roles. 

kibana/sensu roles are not here yet.

**Review by larsks — APPROVED** [review]

File: roles/grafana/files/grafana.repo [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[grafana]
+name=grafana
+baseurl=https://packagecloud.io/grafana/stable/el/6/$basearch
```
Comment (larsks) [inline roles/grafana/files/grafana.repo]: Shouldn't this be `/el/7` (instead of `/6`)?

File: roles/grafana/files/grafana.repo [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+repo_gpgcheck=1
+enabled=1
+gpgcheck=1
+gpgkey=https://packagecloud.io/gpg.key https://grafanarel.s3.amazonaws.com/RPM-GPG-KEY-grafana
+sslverify=1
+sslcacert=/etc/pki/tls/certs/ca-bundle.crt
```
Comment (larsks) [inline roles/grafana/files/grafana.repo]: We shouldn't need to set sslcacert like this (because that's the default).

**Review by larsks — CHANGES_REQUESTED** [review]

**Comment by mrunge** [comment]
@larsks nope. I copied those repository definitions from their upstreams. Especially the "6" in elasticsearch repo is valid. What does this tell us about the repo itself?

**Comment by larsks** [comment]
@mrunge, there is a valid repository at `https://packagecloud.io/grafana/stable/el/7/$basearch`, which is why I was asking.

**Comment by paramite** [comment]
/el/7/ for Grafana seems more legitimate for me too TBH.

**Comment by mrunge** [comment]
dammit. Rebase/push gone wrong.

**Pull request closed without merging**